src/utils/html.js

```diff
@@ -75,8 +75,25 @@ function checkFileTypes (types, url) {
 }
 
 u.isAudioURL = url => checkFileTypes(['.ogg', '.mp3', '.m4a'], url);
-u.isImageURL = url => checkFileTypes(['.jpg', '.jpeg', '.png', '.gif', '.bmp', '.tiff', '.svg'], url);
 u.isVideoURL = url => checkFileTypes(['.mp4', '.webm'], url);
+u.isImageURL = url => {
+    const regex = api.settings.get('image_urls_regex');
+    return regex
+        ? regex.test(url)
+        : checkFileTypes(['.jpg', '.jpeg', '.png', '.gif', '.bmp', '.tiff', '.svg'], url);
+}
+u.isImageDomainAllowed = url => {
+    try {
+        const show_images_inline = api.settings.get('show_images_inline');
+        const is_domains_array = Array.isArray(show_images_inline);
+        if (!is_domains_array) return true;
+        const image_domain = getURI(url).domain();
+        return show_images_inline.includes(image_domain);
```

Comment:
When putting code in a try/catch clause, please put as few lines as possible (ideally only one) there. It should only be the line that you expect to trigger exceptions under certain circumstances.

When there are many lines in a try/catch clause, then other errors in the code might get caught, which can have unexpected results and make debugging harder. It's also harder to see which line is the one where one should expect a potential exception to be raised.

I updated the code here:  https://github.com/conversejs/converse.js/commit/eb016a2aa4f5e9a323afa9e4039bc51c72087452

If this is wrong because the error is expected in a different line, please let me know.